demystifying_sql/demystifying_sql.md

```diff
@@ -0,0 +1,331 @@
+*********
+
+<div class = "answer">
+Relational databases are a data storage solution and they store data in tables, which are in turn organized into rows and columns.  Relational databases use SQL as a way to access data, search for data meeting particular conditions, and connect related data, using columns that tables have in common.  This allows for extracting data in very specific and customized way.
+</div>
+
+***********
+
+</div>
+
+### When to Use SQL
+
+SQL should be used any time you need to access data stored within a relational database and select data that meets your requirements.
+
+Usually, SQL is used for getting custom datasets for export and downstream analysis.  For example, you might use SQL to export data for doing statistical analysis and visualization in other languages like R, Python, or Stata.
+
+If your source data comes from a relational database (like a data warehouse), your major data transformations (like selecting just the columns you care about or selecting only certain rows) should be done using SQL. This ensures that the dataset you export from the relational database is pretty close to the final data you will analyze or visualize.
+
+This is because, especially for large datasets, SQL is a much more efficient tool for large-scale data transformations than your traditional scripting or analytic packages.
+
+<div class = "hint">
+If you think about carving a sculpture out of stone or wood, you can imagine that the rough work of getting rid of big slabs of material that aren't needed can be done with powerful instruments like chainsaws or jackhammers.  Then, when an artist gets close to the shape of the final product, they might switch to smaller tools to give the sculpture its final form.  In this analogy, SQL is the heavy duty tool that gets your data close to its final form.
+</div>
+
+It's a good idea to document your SQL queries and save them, because this allows you to show your steps and give data provenance.  This helps with reproducibility and standardization of your work, and might help provide a place to start for future projects.
+
+### When SQL Isn't the Right Tool
+
+Although SQL is a great tool for organizing your data into meaningful datasets for extraction, it is usually not a great tool to use for your actual analysis work.
+
+Despite having many functions for simple text parsing, SQL is not the tool/language you want to use for advanced NLP (Natural Language Processing) work.  Similarly, SQL has some basic statistical functions, but isn't intended to provide statistical analysis at the level of a statistical programming language like R.
+
+SQL also doesn't have any capabilities to directly support data visualization work.
+
+For all of these "downstream analytics" use cases, you will want to use an actual analytical programing language or tool like R or Python.
```

Comment:
Did you want "programming" here or "programing"? It's spelled the first way in the overview.

Reply:
ohhhh the typoooos will fix